compiler/ballerina-lang/src/main/java/io/ballerina/projects/ModuleContext.java

```diff
@@ -345,6 +351,18 @@ void resolveDependencies(DependencyResolution dependencyResolution) {
         this.moduleDependencies = Collections.unmodifiableSet(moduleDependencies);
     }
 
+    /**
+     *
+     * Returns true if at least one method or type def is connected to one of the root methods of the root pkg.
+     */
+    public boolean isUsed() {
+        return this.bLangPackage().symbol.invocationData.moduleIsUsed;
```

Comment:
I think there is an issue with the logic. If we mark only package is connected to the root based on only there are connected with functions or type definitions it is wrong. There can be other ways as well.
Example:
-  Create a ballerina project
- Add a module
- Define a public variable 
- Then only use that public variable to print the value.

When I do the above steps I got an error. Please check this an update in here. If its working please point the relevant test case as well.

Reply:
I was able to reproduce it. I will look into this and update here. Thank you for pointing this out.

Reply:
Fixed and added a test case in 199a5c1b3200489667996e0c9911eae19b3f461e. The previous behavior was due to not casting the `BIRVariableDcl` to `BIRGlobalVariableDcl` properly. And because of that the correct `PackageID` of the `BIRGlobalVariableDcl` was not fetched. Now packages are marked as used based on `BIRGlobalVariableDcl` usages as well. Thank you again for pointing this out.